src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -265,29 +246,30 @@ public async Task<HttpResponseMessage> SendAsync(CancellationToken cancellationT
             catch (QuicStreamAbortedException ex)
             {
                 // Our stream was reset.
-
                 Exception? abortException = _connection.AbortException;
                 throw new HttpRequestException(SR.net_http_client_execution_error, abortException ?? ex);
             }
             catch (QuicConnectionAbortedException ex)
             {
                 // Our connection was reset. Start shutting down the connection.
-
                 Exception abortException = _connection.Abort(ex);
                 throw new HttpRequestException(SR.net_http_client_execution_error, abortException);
             }
             // It is possible for user's Content code to throw an unexpected OperationCanceledException.
-            catch (OperationCanceledException ex) when (ex.CancellationToken == requestCancellationSource.Token || ex.CancellationToken == _sendContentCts?.Token)
+            catch (OperationCanceledException ex) when (ex.CancellationToken == _requestBodyCancellationSource.Token || ex.CancellationToken == cancellationToken)
             {
                 // We're either observing GOAWAY, or the cancellationToken parameter has been canceled.
                 if (cancellationToken.IsCancellationRequested)
                 {
+                    // in case the HttpContent was duplex, don't cancel _requestBodyCancellationSource
+                    shouldCancelBody = false;
```

Comment:
in that case we might as well skip the condition on the `CancellationTokenRegistration` at the beginning, because the `_requestBodyCancellationSource` will be cancelled indirectly if we hit cancellation via `cancellationToken`. I would actually prefer that.

Reply:
Yes, we should always register `cancellationToken` to the `_requestBodyCancellationSource` (as long as I'm not missing something [here](https://github.com/dotnet/runtime/pull/67954#discussion_r870408423)).

Reply:
good, I reverted that part of the change